smartsim/_core/mli/infrastructure/control/workermanager.py

```diff
@@ -258,95 +295,137 @@ def _on_iteration(self) -> None:
         timings.append(time.perf_counter() - interm)  # timing
         interm = time.perf_counter()  # timing
 
+        reply = InferenceReply()
+
         if not request.raw_model:
             if request.model_key is None:
-                # A valid request should never get here.
-                raise ValueError("Could not read model key")
+                send_failure(request.callback, "Could not find model key or model.")
```

Comment:
Comment for discussion in MLI sync: I just noticed this, but there seems to be two paths for handling errors, one is a call to ``send_failure`` and the other is to call ``exception_handler`` which then calls ``send_failure``.  Is there a way we can align with one method for all cases so that it is consistent what should be done for errors and exceptions?

Reply:
I just added this yesterday. Basically I split it out because with Al's new code there are situations where there is no exception, but we need to return because it's at a state where we can no longer continue. Rather than have the Worker Manager raise an error, we want to just send a response back that there was a failure. Happy to discuss it more in the meeting because I was also unsure what to do in these situations!

Reply:
Actually, I thought I needed to use `send_failure` in more than one spot, but since it's only here I can use `exception_handler` and pass in the `ValueError` so there is only one path for handling error. Fixing this now!

Reply:
`send_failure` has been removed

Reply:
Marking as resolved per our group discussion.